libr/include/r_arch.h

```diff
@@ -0,0 +1,100 @@
+/* radare2 - LGPL - Copyright 2020 - pancake */
+
+#ifndef R2_ARCH_H
+#define R2_ARCH_H
+
+#include <r_util.h>
+#include <r_bind.h>
+#include <r_io.h> // Just for RIOBind which is needed for some plugins
+
+typedef int RArchBits;
+typedef int RArchEndian;
+
+typedef enum r_arch_decode_options_t {
```

Comment:
is it really useful to have all these options as one enum?
What about having separate APIs for each operation?

The reason I'm saying this is that by having a big enum like this, if you pass a RArchInstruction around you never know whether it was initialized with code, data, size, anal, opex, desc or nothing. So you have to do checks all over the places to ensure that the thing you want (e.g. anal) is actually there.

Reply:
already replied this question 3 times above